core/templates/dev/head/i18n.js

```diff
@@ -90,6 +90,11 @@ oppia.controller('I18nFooter', [
         $http.put(siteLanguageUrl, {
           site_language_code: $scope.currentLanguageCode
         });
+      } else {
+        // This is a workaround for the library page as the base url
```

Comment:
This issue only affects users who are not logged in, as per #3551. I thought this might be the easiest workaround.